docs/standard/serialization/system-text-json-supported-collection-types.md

```diff
@@ -0,0 +1,293 @@
+
+::: zone-end
+
+::: zone pivot="dotnet-core-3-1"
+
+| Type                                                        | Serialization | Deserialization |
+|-------------------------------------------------------------|---------------|-----------------|
+| <xref:System.Collections.Concurrent.BlockingCollection%601> | ✔️           | ❌              |
+| <xref:System.Collections.Concurrent.ConcurrentBag%601>      | ✔️           | ❌              |
+| [ConcurrentDictionary\<string, TValue>](xref:System.Collections.Concurrent.ConcurrentDictionary%602) |✔️|✔️|
+| <xref:System.Collections.Concurrent.ConcurrentQueue%601>    | ✔️           | ✔️              |
+| <xref:System.Collections.Concurrent.ConcurrentStack%601> \* | ✔️           | ✔️              |
+
+::: zone-end
+
+\* See [Support round trip for Stack\<T>](system-text-json-converters-how-to.md#support-round-trip-for-stackt).
+
+## System.Collections.ObjectModel namespace
+
+::: zone pivot="dotnet-5-0"
+
+| Type                                                           | Serialization | Deserialization |
+|----------------------------------------------------------------|---------------|-----------------|
+| <xref:System.Collections.ObjectModel.Collection%601>           | ✔️            | ✔️             |
+| [KeyedCollection\<string, TValue>](xref:System.Collections.ObjectModel.KeyedCollection%602) |✔️|❌|
+| <xref:System.Collections.ObjectModel.ObservableCollection%601> | ✔️            | ✔️             |
+| <xref:System.Collections.ObjectModel.ReadOnlyCollection%601>   | ✔️            | ❌             |
+| <xref:System.Collections.ObjectModel.ReadOnlyDictionary%602>   | ✔️            | ❌             |
+| <xref:System.Collections.ObjectModel.ReadOnlyObservableCollection%601> | ✔️    | ❌             |
+
+::: zone-end
+
+::: zone pivot="dotnet-core-3-1"
+
+| Type                                                           | Serialization | Deserialization |
+|----------------------------------------------------------------|---------------|-----------------|
+| <xref:System.Collections.ObjectModel.Collection%601>           | ✔️            | ✔️             |
+| [KeyedCollection\<string, TValue>](xref:System.Collections.ObjectModel.KeyedCollection%602) |✔️|❌|
+| <xref:System.Collections.ObjectModel.ObservableCollection%601> | ✔️            | ✔️             |
+| <xref:System.Collections.ObjectModel.ReadOnlyCollection%601>   | ✔️            | ❌             |
+| [ReadOnlyDictionary\<string, TValue>](xref:System.Collections.ObjectModel.ReadOnlyDictionary%602) |✔️|❌|
+| <xref:System.Collections.ObjectModel.ReadOnlyObservableCollection%601>| ✔️     | ❌             |
+
+::: zone-end
+
+## Custom collections
+
+Any collection type that isn't in one of the preceding namespaces is considered a custom collection. Such types include user-defined types and types defined by ASP.NET Core. For example, <xref:Microsoft.Extensions.Primitives?displayProperty=fullName> is in this group.
```

Comment:
Would a dictionary whose key type isn't string fall under this category?

Reply:
No, because it's in the preceding namespaces, it's just not supported in 3.1.